pkg/packing/packing.go

```diff
@@ -178,14 +178,42 @@ func (*packer) podsMatch(first, second []*v1.Pod) bool {
 
 // sortByResources sorts instance type packings by vcpus and memory resources
 func sortByResources(instances []*Instance) {
-	sort.Slice(instances, func(i, j int) bool {
-		// Euclidean distance from origin using vcpus and memory
-		// sqrt(vcpus[i]^2 + memoryInGiB[i]^2) < sqrt(vcpus[j]^2 + memoryInGiB[j]^2)
-		return math.Sqrt(
-			math.Pow(2, float64(*instances[i].VCpuInfo.DefaultVCpus))+
-				math.Pow(2, float64(*instances[i].MemoryInfo.SizeInMiB)/1024)) <
-			math.Sqrt(
-				math.Pow(2, float64(*instances[j].VCpuInfo.DefaultVCpus))+
-					math.Pow(2, float64(*instances[j].MemoryInfo.SizeInMiB)/1024))
-	})
+	sort.Slice(instances, func(i, j int) bool { return weightOf(instances[i]) < weightOf(instances[j]) })
+}
+
+func weightOf(instance *Instance) float64 {
+	return euclidean(
+		float64(*instance.VCpuInfo.DefaultVCpus),
+		float64(*instance.MemoryInfo.SizeInMiB/1024), // 1 gb = 1 cpu
+		float64(countNvidiaGPUs(instance))*1000,      // Heavily weigh gpus x 1000
+		float64(countAWSNeurons(instance))*1000,      // Heavily weigh neurons x1000
+	)
+}
+
+func euclidean(values ...float64) float64 {
+	sum := float64(0)
+	for _, value := range values {
+		sum += math.Pow(2, value)
```

Comment:
I might be doing the math wrong, but if there's 8 Nvidia GPUs here, won't that result in this being `2^8000`, resulting in overflow?

Reply:
Wow. n^2 not 2^n!

Reply:
not `2^n` factorial either!

Reply:
either eithe eith eit ei e